torchgeo/trainers/detection.py

```diff
@@ -274,6 +290,7 @@ def validation_step(self, *args: Any, **kwargs: Any) -> None:
 
     def on_validation_epoch_end(self) -> None:
         """Logs epoch level validation metrics."""
+        # TODO: why is this method necessary?
```

Comment:
None of our other trainers require special handling for metric logging. Is it only because of https://github.com/Lightning-AI/torchmetrics/pull/1832#issuecomment-1623890714?